python/ray/_raylet.pyx

```diff
@@ -762,6 +762,11 @@ cdef int64_t restore_spilled_objects_handler(
                 "An unexpected internal error occurred while the IO worker "
                 "was restoring spilled objects.")
             logger.exception(exception_str)
+            ray._private.utils.push_error_to_driver(
```

Comment:
We removed it because it is sometimes too verbose (and restoration failure should be transparently handled), but I added back for debugging purpose. We can remove anytime if this makes verbose error messages again

Reply:
could we only send this when debug mode is on?

Reply:
should be easy to do that! (just read env var)

Reply:
Addressed. lmk if you think we need to add a test (I didn't do that because it is a little tricky to do that, but it is doable techincally)